apis/Google.Cloud.Diagnostics.AspNet/Google.Cloud.Diagnostics.AspNet.IntegrationTests/Trace/TraceTest.cs

```diff
@@ -89,13 +89,16 @@ private void BlockUntilClockTick()
         /// <summary>
         /// Gets a trace that contains a span with the given name.
         /// </summary>
-        private async Task<TraceProto> GetTrace(string spanName)
+        /// <param name="expectTrace">True if the trace is expected to exist.  This is used
+        ///     to minimize RPC calls.</param>
+        private async Task<TraceProto> GetTrace(string spanName, bool expectTrace = true)
         {
-            double sleepTime = 0;
-            while (sleepTime < _timeout.TotalMilliseconds)
+            double totalSleepTime = 0;
```

Comment:
Why are we using numbers at all here? Just keep it all in TimeSpan - you can add them together, compare them, sleep etc. Eschew numbers!